### PR TITLE
RouterTabs Docs

### PR DESCRIPTION
**Pull request by thomasdax98**
depends on https://github.com/vivid-planet/comet-admin/pull/633

**Topics:**
- Basic `Tabs`
- Differences between Comet Admin `Tabs` and `MuiTabs`
- Basic `RouterTabs`
- `RouterTabs` in combination with `Stack` (special case:  multiple nested `StackSwitch`)

Check out here: https://deploy-preview-620--comet-admin.netlify.app/?path=/story/docs-components-tabs-tabs--page

TODO:

- [x] Example for `forceRender`

### CODE REVIEW

**Review by jamesricky — APPROVED** [review]

**Comment by thomasdax98** [comment]
@nsams 
> > only two RouterTabs are shown simultaneously at each point in time
> 
> Is that a bug or a feature we intentionally implemeted?

As far as I can tell, that's an intentional feature. It's the sole reason that the `shouldShowTabBar` variable exists (https://github.com/vivid-planet/comet-admin/blob/d4b14027c024a76c70f81a78f0d16b4d3bbef49f/packages/admin/src/tabs/RouterTabs.tsx#L97) 

According to Git Blame it was implemented by @jamesricky in 2021. Maybe he can tell us more?

> PS: please try to make more PRs - separating independent bugfixes, refactoring and documentation should be done if possible

Okay, will do

**Comment by jamesricky** [comment]
> According to Git Blame it was implemented by @jamesricky in 2021. Maybe he can tell us more?

@thomasdax98 I just renamed the file or something. It was initially implemented by @nsams in https://github.com/vivid-planet/comet-admin/commit/e6cc5edb8b44aed8ac74f5cc64fe18c89aa055d8

**Comment by nsams** [comment]
> @thomasdax98 I just renamed the file or something. It was initially implemented by @nsams in https://github.com/vivid-planet/comet-admin/commit/e6cc5edb8b44aed8ac74f5cc64fe18c89aa055d8

:)

ok - that feature: afair that feature exists to hide the outer tabs once you go into a nested StackPage. (there is no limit on two tabs or so) That was initially implemented for glappvisittooladmin (still testable there)

**Comment by thomasdax98** [comment]
> > @thomasdax98 I just renamed the file or something. It was initially implemented by @nsams in [e6cc5ed](https://github.com/vivid-planet/comet-admin/commit/e6cc5edb8b44aed8ac74f5cc64fe18c89aa055d8)
> 
> :)
> 
> ok - that feature: afair that feature exists to hide the outer tabs once you go into a nested StackPage. (there is no limit on two tabs or so) That was initially implemented for glappvisittooladmin (still testable there)

@nsams I rephrased it to 

https://github.com/vivid-planet/comet-admin/pull/620/files#diff-e9fd0ba5de0a403d13a2135433c3ac60a5fa2d04d9d6aa0354ee37434eae9953R30-R32

File: packages/admin-stories/src/docs/components/RouterTabs/02_RouterTabs.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+-   `tabComponent` for passing a custom tab component. Per default, [Material UI's `Tab` component](https://v4.mui.com/api/tab/) is used
+-   `tabsProps` for passing props to [Material UI's `Tabs` component](https://v4.mui.com/api/tabs/)
+
+As children, an array of `RouterTab` elements should be passed to `RouterTabs`.
+Each `RouterTab` needs:
+
+-   a `label` that is shown in the tab bar
+-   a unique `path` of the tab that is attached to the URL
+-   optionally a `forceRender` flag.
```
Comment (johnnyomair) [inline packages/admin-stories/src/docs/components/RouterTabs/02_RouterTabs.stories.mdx:20]: Maybe we could add an example where this prop should be used. I can't think of a use case immediately.

File: packages/admin/src/tabs/RouterTabs.tsx [hunk 2]
```diff
@@ -63,63 +71,58 @@ function RouterTabsComponent({
...
 
+    React.useEffect(() => {
+        if (stackApi && stackSwitchApi) {
+            // When inside a Stack show only the last TabBar
+            const ownSwitchIndex = stackSwitchApi.id ? stackApi.switches.findIndex((i) => i.id === stackSwitchApi.id) : -1;
+            const nextSwitchShowsInitialPage = stackApi.switches[ownSwitchIndex + 1] && stackApi.switches[ownSwitchIndex + 1].isInitialPageActive;
+
+            setShouldShowTabBar(ownSwitchIndex === stackApi.switches.length - (nextSwitchShowsInitialPage ? 2 : 1));
+        }
+    }, [stackApi, stackSwitchApi]);
```
Comment (johnnyomair) [inline packages/admin/src/tabs/RouterTabs.tsx:82]: Previously, this was not a state, but simply a variable. Why did you change that? Doing so causes an additional render when `stackApi` or `stackSwitchApi` changes.

File: packages/admin-stories/src/docs/components/RouterTabs/02_RouterTabs.stories.mdx (hunk 1 above)
Comment (thomasdax98) [reply packages/admin-stories/src/docs/components/RouterTabs/02_RouterTabs.stories.mdx:20]: I also couldn't think of a sensible example use case.
At first, I thought SEO purposes, but that doesn't make sense for an admin application... 🤔

Comment (thomasdax98) [reply packages/admin-stories/src/docs/components/RouterTabs/02_RouterTabs.stories.mdx:20]: **Update:**

It's necessary for Forms that stretch over multiple Tabs because FinalForm "forgets" Fields that aren't rendered anymore. Then, form validation and dirty state don't work anymore

Comment (thomasdax98) [reply packages/admin-stories/src/docs/components/RouterTabs/02_RouterTabs.stories.mdx]: @johnnyomair I added a section about the `forceRender` prop

**Comment by johnnyomair** [comment]
@thomasdax98 I believe `forceRender` prop also applies to "normal" tabs, for instance when using a form spread across multiple tabs. It would make sense to move the explanation of the prop to the tabs page.

**Review by johnnyomair — APPROVED** [review]
